packages/sg-std/src/fees.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{create_fund_community_pool_msg, NATIVE_DENOM};
+use cosmwasm_std::{coin, coins, BankMsg, CosmosMsg, Decimal, Env, MessageInfo};
+use cw_utils::{must_pay, PaymentError};
+use thiserror::Error;
+
+const FEE_BURN_PERCENT: u64 = 50;
+
+pub fn handle_fee<T>(
+    env: Env,
+    info: &MessageInfo,
+    expected_fee: u128,
+) -> Result<Vec<CosmosMsg<T>>, FeeError> {
+    let payment = must_pay(info, NATIVE_DENOM)?;
+    if payment.u128() != expected_fee {
+        return Err(FeeError::InvalidFee {});
+    }
+
+    // calculate the fee to burn
+    let burn_percent = Decimal::percent(FEE_BURN_PERCENT);
+    let burn_fee = payment * burn_percent;
+    let burn_coin = coin(burn_fee.u128(), NATIVE_DENOM);
+    // send fee to contract to be burned
+    let send_fee_msg = BankMsg::Send {
+        to_address: env.contract.address.to_string(),
+        amount: vec![burn_coin.clone()],
+    };
+    // burn half the fee
+    let fee_burn_msg = BankMsg::Burn {
+        amount: vec![burn_coin],
+    };
```

Comment:
So we're sending it to a contract to be burned and burning it?

Reply:
Is one of these not needed?

Reply:
Yeah exactly. Know of a better way? Another way is to send it to an address where the private key is destroyed.